pkg/render/common/components/components.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2022 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package components
+
+import (
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/components"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// ApplyDaemonSetOverrides applies the overrides to the given DaemonSet.
+// Note: overrides must not be nil pointer.
+func ApplyDaemonSetOverrides(ds *appsv1.DaemonSet, overrides components.DaemonSetOverride) *appsv1.DaemonSet {
+	// Catch if caller passes in an explicit nil.
+	if overrides == nil {
+		return ds
+	}
+
+	var metadata *operatorv1.Metadata
+	var minReadySeconds *int32
+	var podTemplateMetadata *operatorv1.Metadata
+
+	var affinity *corev1.Affinity
+	var tolerations []corev1.Toleration
+	var nodeSelector map[string]string
+	var initContainers []operatorv1.Container
```

Comment:
This code will be a lot easier to follow if we move the declaration of the variables closer to where we use them. 

Also, I don't think we need to have a separate variable declaration / population any more. Could probably even do something like this:

```go
if ics := overrides.GetInitContainers(); ics != nil {
  // Handle init container overrides.
}
```

Reply:
Fixed